File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/tag/model/Tag.java
package com.side.tiggle.domain.tag.model;

import com.side.tiggle.global.common.model.BaseEntity;
import lombok.*;

import javax.persistence.*;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "tags")
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "defaults", nullable = false)
    private boolean defaults;

    @Builder
    public Tag(String name, boolean defaults) {
        this.name = name;
        this.defaults = defaults;

    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/tag/model/Tag.kt
package com.side.tiggle.domain.tag.model

import com.side.tiggle.domain.member.model.Member
import javax.persistence.*

@Entity
@Table(name = "tags")
class Tag(
    @Column(name = "name", nullable = false)
    var name: String,
    @Column(name = "defaults", nullable = false)
    val defaults: Boolean = false
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    var member: Member? = null
}